src/software/ai/navigator/navigator.cpp

```diff
@@ -44,56 +49,54 @@ void Navigator::visit(const KickIntent &kick_intent)
 
 void Navigator::visit(const MoveIntent &move_intent)
 {
+    std::vector<Point> path_points;
+
     Point start =
         this->world.friendlyTeam().getRobotById(move_intent.getRobotId())->position();
     Point dest = move_intent.getDestination();
 
     std::vector<Obstacle> obstacles =
         createCurrentObstacles(move_intent.getAreasToAvoid(), move_intent.getRobotId());
 
-    auto path_planner =
-        std::make_unique<ThetaStarPathPlanner>(this->world.field(), obstacles);
-
-    auto path_points = path_planner->findPath(start, dest);
+    try
+    {
+        path_points = std::get<std::vector<Point>>(
+            path_planner_->findPath(start, dest, this->world.field(), obstacles));
+    }
+    catch (const std::bad_variant_access &)
+    {
+        // variant either monostate or Curve, so no valid path
```

Comment:
lets log something here so we know why the robots stopped

Reply:
This will likely flood the log, because we often try to move to where we already are.

Reply:
catching bad_variant_access means we found no path (we don't have curves atm), and the variant is of type monostate (null type for variants)

Reply:
ok scratch that, I thought this was the case the planner couldn't find any path and the robots just stop

Reply:
Hmm, I'd rather return an empty vector in this case, just be explicit that there's no path. I don't think just returning a monostate like this is very elegant or explicit.

Reply:
It's the idea of std::optional for the vector of points but for variants. Previously, we didn't return empty vector, we returned std::nullopt

Reply:
If we were to return a curve here and not handle it, the _last_ thing I'd want to do is just suppress the error. Realistically if we're returning the wrong type (ie not handling curves and one is returned), I think the exception should just be propagated up.

Reply:
Ok

Reply:
Done

Reply:
LGTM, again will let @akhilveeraghanta close his comment